Source/Core/DiscIO/NANDImporter.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+
+namespace DiscIO
+{
+class NANDImporter final
+{
+public:
+  static NANDImporter& GetInstance()
+  {
+    static NANDImporter instance;
+    return instance;
+  }
+  void ImportNANDBin(const std::string& path_to_bin);
+  void ExtractCertificates(const std::string& nand_root);
+
+private:
+  struct NANDFSTEntry;
+
+  bool ReadNANDBin(const std::string& path_to_bin);
+  void FindSuperblock();
+  std::string GetPath(const NANDFSTEntry& entry, const std::string& parent_path);
+  void ProcessEntry(u16 entry_number, const std::string& parent_path);
+  void ProcessFile(const NANDFSTEntry& entry, const std::string& parent_path);
+  void ProcessDirectory(const NANDFSTEntry& entry, const std::string& parent_path);
+  void ExportKeys(const std::string& nand_root);
+
+#pragma pack(push, 1)
+  struct NANDFSTEntry
```

Comment:
I should have pointed this out before but struct definitions should come before functions.